backend/src/apiserver/common/utils.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strings"
+)
+
+// CreateArtifactPath creates artifact resource path.
+func CreateArtifactPath(runID string, nodeID string, artifactName string) string {
+	return fmt.Sprintf("runs/%s/nodes/%s/artifacts/%s", runID, nodeID, artifactName)
+}
+
+// Fetches parent resource ids from a resource name.
+// Supports namespaces, pipelines, pipeline versions, experiments, runs, recurring runs.
+//
+// Example:
+//
+//	namespaces/ns1/pipelines/p1/versions/p1.1 results in
+//	  Namespace         : ns1
+//	  PipelineId        : p1
+//	  PipelineVersionId : p1.1
+func ParseResourceIdsFromFullName(p string) map[string]string {
+	results := map[string]string{
+		"Namespace":         "",
+		"ExperimentId":      "",
+		"PipelineId":        "",
+		"PipelineVersionId": "",
+		"RunId":             "",
+		"RecurringRunId":    "",
+		"ArtifactId":        "",
+		"ExecutionId":       "",
+	}
+	names := strings.Split(p, "/")
+	i := 0
+	for i < len(names) {
+		switch strings.ToLower(names[i]) {
+		case "namespaces", "namespace":
+			results["Namespace"] = names[i+1]
+		case "pipelines", "pipeline":
+			results["PipelineId"] = names[i+1]
+		case "versions", "version", "pipelineversions", "pipelineversion", "pipeline_versions", "pipeline_version":
```

Comment:
Good point. Thank you! As we use the shortcuts `versions` in v2beta1 API, I propose to keep it. The singular `version` was added for consistency with other types (in v1beta1, we often use singular `namespace` instead of `namespaces`, and so on). The other variations are coming from the intention to include full resource names in addition to the shortcuts. Which options do you think are redundant? 

/cc @Linchin

Reply:
I don't think we used the same resource paths for v1 and v2, so how can we use the same function for both APIs? Are these resource names constructed internally?

Reply:
I probably don't have a full context. Can you please elaborate what are the differences between v1 and v2?